sigstore/_internal/rekor/client.py

```diff
@@ -219,14 +194,57 @@ def post(
             },
         }
 
-        resp: requests.Response = self.session.post(self.url, data=json.dumps(data))
+        resp: requests.Response = self.session.post(self.url, json=data)
         try:
             resp.raise_for_status()
         except requests.HTTPError as http_error:
             raise RekorClientError from http_error
 
         return RekorEntry.from_response(resp.json())
 
+    @property
+    def retrieve(self) -> RekorEntriesRetrieve:
+        return RekorEntriesRetrieve(
+            urljoin(self.url, "retrieve/"), session=self.session
+        )
+
+
+class RekorEntriesRetrieve(Endpoint):
+    def post(
+        self,
+        b64_artifact_signature: str,
+        sha256_artifact_hash: str,
+        b64_cert: str,
+    ) -> RekorEntry:
+        data = {
+            "entries": [
+                {
+                    "kind": "hashedrekord",
+                    "apiVersion": "0.0.1",
+                    "spec": {
+                        "signature": {
+                            "content": b64_artifact_signature,
+                            "publicKey": {"content": b64_cert},
+                        },
+                        "data": {
+                            "hash": {
+                                "algorithm": "sha256",
+                                "value": sha256_artifact_hash,
+                            }
+                        },
+                    },
+                }
+            ]
+        }
+
+        resp: requests.Response = self.session.post(self.url, json=data)
+        try:
+            resp.raise_for_status()
+        except requests.HTTPError as http_error:
+            raise RekorClientError(resp.json()) from http_error
+
+        return RekorEntry.from_response(resp.json()[0])
```

Comment:
I'm not totally confident that entries are received in earliest entry order...
I think in practice it actually goes active -> inactives
https://github.com/sigstore/rekor/blob/01c7b55405527d40de664d81696b5e78e7abbb0d/pkg/api/entries.go#L423

I can't decide if it make sense for this to return the array of RekorEntry's or to be handling the  parsing, ordering, and returning of a single internally. Maybe the latter is better for use-ability outside the package... but fairly complex for a client.

Reply:
> I can't decide if it make sense for this to return the array of RekorEntry's or to be handling the parsing, ordering, and returning of a single internally. Maybe the latter is better for use-ability outside the package... but fairly complex for a client.

Yeah, I think for now I'll make the client do it internally (especially since these are all currently internal API details for `sigstore-python`).

Longer term it might make sense to make a separate Rekor API client that `sigstore-python` depends on, which in turn should isolate these concerns. But that's not a project for today 🙂